### ISSUE TITLE
reconsider the current logic related to `babel.config.js`

### ISSUE THREAD

**Issue by justinhelmer** [vuejs/vue-jest#109]
The change from [this commit](https://github.com/vuejs/vue-jest/commit/0283ef99fc7756c9ec60d836e6b97a31b773439e) https://github.com/vuejs/vue-jest/commit/0283ef99fc7756c9ec60d836e6b97a31b773439e introduced the concept of `babel.config.js`. However, the current logic assumes the file exports a [configuration object](https://github.com/vuejs/vue-jest/commit/0283ef99fc7756c9ec60d836e6b97a31b773439e#diff-ebf5102d63a5b2bed23f3b9499bbed83R19).

The [Babel Configuration Docs](https://babeljs.io/docs/en/configuration) are now promoting the use of `babel.config.js` for project-wide configuration - but the format of the file is to [export a function](https://babeljs.io/docs/en/configuration#babelconfigjs).

I submitted another PR (#111) to help provide support for this new config file type (currently blocked by #108), and it should not conflict with the current logic. However, there may be unintended consequences of the current implementation if projects use the `configFile` or `root` options to load their [project-wide configuration](https://babeljs.io/docs/en/config-files#project-wide-configuration). In this scenario, babel will be assuming that `babel.config.js` exports a function, which could be seen as inconsistent with [the current logic](https://github.com/vuejs/vue-jest/commit/0283ef99fc7756c9ec60d836e6b97a31b773439e#diff-ebf5102d63a5b2bed23f3b9499bbed83R19).

**Comment by eddyerburgh** [comment]
This should be fixed by https://github.com/vuejs/vue-jest/pull/127, to be release in v4. We'll use babel-jest to resolve babel files. 

Before v4 is released, I'll update vue-jest to export a `createTransformer` method that gives users a way to customize babel-jest.

**Comment by Globegitter** [comment]
I am just trying out the latest v4 beta and for me the option:

```
  globals: {
    'vue-jest': {
      transform: {
        'js': './custom/babel-jest.js',
      },
    },
  },
```

is working well. As we already have a custom `babel-jest` to find the babel config at our custom location, so we can just reuse that.
